Histogram.py

```diff
@@ -0,0 +1,169 @@
+
+    # PIDs of dark daughter
+    dark_daughter = pid[d1[is_dark]] 
+    is_dark_daughter = ak.zeros_like(dark_daughter)
+
+    for dsid in stable_particle_ids:
+        is_dark_daughter = is_dark_daughter | (np.abs(dark_daughter)==dsid)
+
+    stability = [None]*len(events)
+    for i in range(len(pid)): 
+        if ak.sum(is_dark[i])!=0:
+            stability[i] = ak.sum(is_dark_daughter[i])/ak.sum(is_dark[i])
+
+    #stability = [ak.sum(is_dark_daughter[i])/ak.sum(is_dark[i]) for i in range(len(pid)) if ak.sum(is_dark[i])!=0]
+
+    # Add the invisible fraction to the events
+    events["stable_invisible_fraction"] = stability
+
+    # store modified events array
+    Events = events
+    
+    
+    # Histogram
+
+    # Creating hist objects
+    hists_mt = fill_hist("MT",25,0,1500,r"$m_{\text{T}}$ [GeV]",Events)
+    hists_pt = fill_hist("Jet_pt",50,0,1000,r"$p_{\text{T}}$ [GeV]",Events)
+    hists_pt1 = fill_hist("Jet1_pt",50,0,1000,r"$Jet_1\_p_{\text{T}}$ [GeV]",Events)
+    hists_pt2 = fill_hist("Jet2_pt",50,0,1000,r"$Jet_2\_p_{\text{T}}$ [GeV]",Events)
+    hists_met = fill_hist("MET",50,0,1000,r"$p_{\text{T},miss}$ [GeV]",Events)
+    hists_eta = fill_hist("Jet_eta",50,-10,10,r"$Jet_{\eta}$ [GeV]",Events)
+    hists_eta1 = fill_hist("Jet1_eta",50,-6,6,r"$\eta_{Jet1}$",Events)
+    hists_eta2 = fill_hist("Jet2_eta",50,-6,6,r"$\eta_{Jet2}$",Events)
+    hists_deltaeta = fill_hist("DeltaEta",35,0,8.0,r"$\Delta\eta$",Events)
+    hists_phi = fill_hist("Jet_phi",25,-4,4,r"$Jet_{\phi}$",Events)
+    hists_phi1 = fill_hist("Jet1_phi",25,-4,4,r"$\phi_{Jet1}$",Events)
+    hists_phi2 = fill_hist("Jet2_phi",25,-4,4,r"$\phi_{Jet2}$",Events)
+    hists_deltaphi = fill_hist("DeltaPhi",20,0,3.5,r"$\Delta\Phi$",Events)
+    hists_mass = fill_hist("Jet_mass",50,0,2300,r"$Jet_{mass}$ [GeV]",Events)
+    hists_mass1 = fill_hist("Jet1_mass",50,0,250,r"$Jet1_{mass}$ [GeV]",Events)
+    hists_mass2 = fill_hist("Jet2_mass",50,0,250,r"$Jet2_{mass}$ [GeV]",Events)
+    hists_deltaphi1 = fill_hist("DeltaPhi_MET_Jet1",25,0,3.5,r"$\Delta\phi\_P_{t,miss}\_Jet_1$",Events)
+    hists_deltaphi2 = fill_hist("DeltaPhi_MET_Jet2",25,0,3.5,r"$\Delta\phi\_P_{t,miss}\_Jet_2$",Events)
+    hists_if = fill_hist("stable_invisible_fraction",25,0,1,r"stable_invisible_fraction",Events)
+
+    hist_dict = {
+        "Jet_mt": hists_mt,
+        "Jet_pt": hists_pt,
+        "Jet1_pt": hists_pt1,
+        "Jet2_pt": hists_pt2,
+        "MET": hists_met,
+        "Jet_eta": hists_eta,
+        "Jet1_eta": hists_eta1,
+        "Jet2_eta": hists_eta2,
+        "Jet_phi": hists_phi,
+        "Jet1_phi": hists_phi1,
+        "Jet2_phi": hists_phi2,
+        "Jet_mass": hists_mass,
+        "Jet1_mass": hists_mass1,
+        "Jet2_mass": hists_mass2,
+        "DeltaEta": hists_deltaeta,
+        "DeltaPhi": hists_deltaphi,
+        "DeltaPhi_MET_Jet1": hists_deltaphi1,
+        "DeltaPhi_MET_Jet2": hists_deltaphi2,
+        "stable_invisible_fraction": hists_if
+    }
```

Comment:
These two sections should be combined, i.e. fill the dictionary immediately rather than assigning temporary variables. (The goal is to minimize how many places you need to edit the code to add a new histogram, so it's as easy as possible)